frontend/src/main/scala/bloop/exec/ProcessConfig.scala

```diff
@@ -74,23 +101,29 @@ case class InProcess(classpath: Array[AbsolutePath]) extends ProcessConfig {
  * Configuration to start a new JVM to execute Java code.
  */
 case class Fork(classpath: Array[AbsolutePath], javaEnv: JavaEnv) extends ProcessConfig {
-  override def runMain(className: String, args: Array[String], logger: Logger): Int = {
+  override def runMain(className: String,
+                       args: Array[String],
+                       logger: Logger,
+                       extraClasspath: Array[AbsolutePath] = Array.empty): Int = {
+    val fullClasspath = classpath ++ extraClasspath
+
     val java = javaEnv.javaHome.resolve("bin").resolve("java")
-    val classpathOption = "-cp" :: classpath.map(_.syntax).mkString(pathSeparator) :: Nil
+    val classpathOption = "-cp" :: fullClasspath.map(_.syntax).mkString(pathSeparator) :: Nil
     val appOptions = className :: args.toList
     val cmd = java.syntax :: javaEnv.javaOptions.toList ::: classpathOption ::: appOptions
 
     logger.debug(s"Running '$className' in a new JVM.")
     logger.debug(s"  java_home   = '${javaEnv.javaHome}'")
     logger.debug(s"  javaOptions = '${javaEnv.javaOptions.mkString(" ")}'")
-    logger.debug(s"  classpath   = '${classpath.map(_.syntax).mkString(pathSeparator)}'")
+    logger.debug(s"  classpath   = '${fullClasspath.map(_.syntax).mkString(pathSeparator)}'")
     logger.debug(s"  command     = '${cmd.mkString(" ")}'")
 
     val processBuilder = new ProcessBuilder(cmd: _*)
     val process = processBuilder.start()
     val processLogger = new ProcessLogger(logger, process)
     processLogger.start()
     val exitCode = process.waitFor()
+    logger.debug(s"Forked JVM exited with code: $exitCode")
```

Comment:
As far as I see, we're not supporting to run tests and main with a different java home, as sbt allows you to, right? This is only an appreciation, but we may want to support this out of the box in a follow up PR since I think it's quite a popular request: lots of people are stuck in JDK6 and JDK7, and they want to try their 2.11 and 2.10 artifacts there.

Reply:
`sbt-bloop` already extracts the java home from the config, and it's passed to the `JavaEnv`, so it should work. What do you have in mind? Having a set of different java homes from which we can select before running?

Reply:
If the Java home can be set by the user it’s fine.